tests/test_envs.py

```diff
@@ -49,90 +38,41 @@ def assert_equal_rollout(rollout_a, rollout_b):
     assert info_a == info_b
 
 
-@pytest.mark.parametrize("env_name", ENV_NAMES)
-def test_seed(env, env_name):
-  env.action_space.seed(0)
-  actions = [env.action_space.sample() for _ in range(10)]
-
-  # With the same seed, should always get the same result
-  seeds = env.seed(42)
-  assert isinstance(seeds, list)
-  assert len(seeds) > 0
-  rollout_a = rollout(env, actions)
-
-  env.seed(42)
-  rollout_b = rollout(env, actions)
-
-  assert_equal_rollout(rollout_a, rollout_b)
-
-  # For non-deterministic environments, if we try enough seeds we should
-  # eventually get a different result. For deterministic environments, all
-  # seeds will produce the same starting state.
-  same_obs = True
-  for i in range(20):
-    env.seed(i)
-    new_rollout = rollout(env, actions)
-    for step_a, step_new in zip(rollout_a, new_rollout):
-      obs_a = step_a[0]
-      obs_new = step_new[0]
-      if np.any(obs_a != obs_new):
-        same_obs = False
-        break
-    if not same_obs:
-      break
-
-  is_deterministic = matches_list(env_name, DETERMINISTIC_ENVS)
-  assert same_obs == is_deterministic
-
-
-@pytest.mark.parametrize("env_name", ENV_NAMES)
-def test_premature_step(env):
-  if hasattr(env, 'sim') and hasattr(env, 'model'):  # pragma: no cover
-    # We can't use isinstance since importing mujoco_py will fail on
-    # machines without MuJoCo installed
-    pytest.skip("MuJoCo environments cannot perform this check.")
-
-  act = env.action_space.sample()
-  with pytest.raises(Exception):  # need to call env.reset() first
-    env.step(act)
+@pytest.fixture
+def env(env_name):
+  try:
+    env = gym.make(env_name)
+  except gym.error.DependencyNotInstalled as e:  # pragma: no cover
+    if e.args[0].find('mujoco_py') != -1:
+      pytest.skip("Requires `mujoco_py`, which isn't installed.")
+    else:
+      raise
+  return env
 
 
 @pytest.mark.parametrize("env_name", ENV_NAMES)
-def test_model_based(env):
-  """Smoke test for each of the ModelBasedEnv methods with basic type checks."""
-  if not hasattr(env, 'state_space'):  # pragma: no cover
-    pytest.skip("This test is only for subclasses of ModelBasedEnv.")
-
-  state = env.initial_state()
-  assert env.state_space.contains(state)
-
-  action = env.action_space.sample()
-  new_state = env.transition(state, action)
-  assert env.state_space.contains(new_state)
-
-  reward = env.reward(state, action, new_state)
-  assert isinstance(reward, float)
+class TestEnvs:
```

Comment:
I'm curious why you chose to add a class wrapper here. Does this serve any purpose besides adding another layer of hierarchy in the test logs?

Reply:
All the methods are parameterized by `env_name`, so it seems a bit cleaner to parameterize the class rather than individual methods.